pkg/streamingpromql/operator/pool.go

```diff
@@ -19,6 +19,10 @@ var (
 		return make([]promql.FPoint, 0, size)
 	})
 
+	hPointSlicePool = pool.NewBucketedPool(1, maxExpectedPointsPerSeries, 10, func(size int) []promql.HPoint {
```

Comment:
Heads up that we should change `10` here to use `pointsPerSeriesBucketFactor` once https://github.com/grafana/mimir/pull/8096 is merged.